### PR TITLE
Fix distance_Zq with get_representative

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR fixes all errors form changes in the math crate, regarding the change from distance_Zq and get_representative.

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases
- [x] I provided an intuition regarding how certain inputs have to be set

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/construction/hash/sha256.rs [hunk 1]
```diff
@@ -251,7 +251,7 @@ impl HashInto<MatPolynomialRingZq> for HashMatPolynomialRingZq {
         let highest_deg = self.modulus.get_degree();
         let embedding =
             hash_to_mat_zq_sha256(m, self.rows * highest_deg, self.cols, self.modulus.get_q())
-                .get_mat();
+                .get_representative_least_nonnegative_residue();
```
Comment (jnsiemer) [inline src/construction/hash/sha256.rs:254]: I don't think this is a speaking name. "get_mat" was more intuitive, wasn't it? We should think about changing this back. This way, it seems more precise but also way harder to locate / find the function, which only outputs a clone of the original object.

**Review by jnsiemer — APPROVED** [review]
